salt/loader.py

```diff
@@ -1177,10 +1177,10 @@ def refresh_file_mapping(self):
         # The files are added in order of priority, so order *must* be retained.
         self.file_mapping = salt.utils.odict.OrderedDict()
 
-        for mod_dir in self.module_dirs:
+        for mod_dir in sorted(self.module_dirs):
```

Comment:
I created https://github.com/saltstack/salt/issues/40517 in order not to forget about this